libnd4j/include/ops/declarable/platform/cudnn/maxpool2d.cu

```diff
@@ -123,9 +136,12 @@ PLATFORM_CHECK(maxpool2d_bp, ENGINE_CUDA) {
 
     const auto goodType = input->dataType() == DataType::DOUBLE || input->dataType() == DataType::FLOAT32 || input->dataType() == DataType::HALF || input->dataType() == DataType::INT32;
 
-    return goodType && (input->dataType() == gradO->dataType())
-                    && (input->dataType() == gradI->dataType())
-                    && shape::haveSameShapeAndStrides(input->shapeInfo(), gradI->shapeInfo());
+    return  goodType && (input->dataType() == gradO->dataType())
```

Comment:
Mind adding a minor comment paragraph explaining the issues found here so we know why we're defaulting to our implementation in certain cases?

Reply:
done